File: microservices/review/service/src/main/java/fr/edjaz/microservices/core/review/ReviewServiceApplication.java
package fr.edjaz.microservices.core.review;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import reactor.core.scheduler.Scheduler;
import reactor.core.scheduler.Schedulers;

import java.util.concurrent.Executors;

@SpringBootApplication
@ComponentScan("fr.edjaz")
@Slf4j
public class ReviewServiceApplication {


	private final Integer connectionPoolSize;

	@Autowired
	public ReviewServiceApplication(
		@Value("${spring.datasource.maximum-pool-size:10}")
		Integer connectionPoolSize
	) {
		this.connectionPoolSize = connectionPoolSize;
	}

    @Bean
    public Scheduler jdbcScheduler() {
        LOG.info("Creates a jdbcScheduler with connectionPoolSize = " + connectionPoolSize);
        return Schedulers.fromExecutor(Executors.newFixedThreadPool(connectionPoolSize));
    }

	public static void main(String[] args) {
		ConfigurableApplicationContext ctx = SpringApplication.run(ReviewServiceApplication.class, args);

		String mysqlUri = ctx.getEnvironment().getProperty("spring.datasource.url");
		LOG.info("Connected to MySQL: " + mysqlUri);
	}
}

File: microservices/review/service/src/main/kotlin/fr/edjaz/microservices/core/review/ReviewServiceApplication.kt
package fr.edjaz.microservices.core.review

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cloud.client.discovery.EnableDiscoveryClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import reactor.core.scheduler.Scheduler
import reactor.core.scheduler.Schedulers
import java.util.concurrent.Executors


@ComponentScan("fr.edjaz")
@EnableDiscoveryClient
@SpringBootApplication
class ReviewServiceApplication {
  companion object {
    @Suppress("JAVA_CLASS_ON_COMPANION")
    @JvmStatic
    val logger = LoggerFactory.getLogger(javaClass.enclosingClass)
  }

  private var connectionPoolSize: Int

  @Autowired
  constructor(@Value("\${spring.datasource.maximum-pool-size:10}") connectionPoolSize: Int) {
    this.connectionPoolSize = connectionPoolSize
  }


  @Bean
  fun jdbcScheduler(): Scheduler? {
    logger.info("Creates a jdbcScheduler with connectionPoolSize = $connectionPoolSize")
    return Schedulers.fromExecutor(Executors.newFixedThreadPool(connectionPoolSize))
  }


}

fun main(args: Array<String>) {
  val ctx = runApplication<ReviewServiceApplication>(*args)
  val mysqlUri = ctx.environment.getProperty("spring.datasource.url")
  ReviewServiceApplication.logger.info("Connected to MySQL: $mysqlUri")
}
